.github/workflows/build.yml

```diff
@@ -113,30 +99,23 @@ jobs:
 
     steps:
     - name: Checkout
-      uses: actions/checkout@v1
+      uses: actions/checkout@v2
 
     - name: Checkout dateutil
-      uses: actions/checkout@v1
+      uses: actions/checkout@v2
       with:
         repository: dateutil/dateutil
         path: dateutil
         ref: 'master'
 
     - name: Set up Python
-      uses: actions/setup-python@v1
+      uses: actions/setup-python@v2
 
-    - name: Install dependencies
-      run: pip install toml pep517 packaging importlib_metadata
-
-    - name: Install python2 dependencies
-      if: ${{ matrix.python == 2.7 }}
-      run: pip install typing
-
-    - name: Install build backend
-      run: pip install setuptools wheel
+    - name: install build
+      run: pip install . setuptools wheel
```

Comment:
It's semi-related. The ``build.yaml`` was broken once I moved to the src layout because this again only worked because the CWD is auto-added to the sys.path.

Reply:
Then use `PYTHONPATH` intead of `pip`.

Reply:
That would be a bad idea. We want to validate that things work with the installed package, not that it works on the local source tree. People, in general, will use this package as installed in package format, not by copying the build files and then hacking it on the sys.path. Hacking the sys.path is exactly that allowed for you to not notice until now that the source distribution is broken. Furthermore, installing the package itself means now you not need to install the build dependencies separately, those get pulled in automatically.

Reply:
I also want to validate it works in-tree, that's how I would use it when bootstrapping.

Reply:
Fair, but that's then separate tests. And probably better done by moving these as integration tests. I'd suggest we make such exhaustive tests there, we can run builds both as installed and inside a clear virtual environment that has no packages within, but sets the python path accordingly.